aiosmtpd/smtp.py

```diff
@@ -197,18 +190,33 @@ def _handle_client(self):
             line = yield from self._reader.readline()
             try:
                 # XXX this rstrip may not completely preserve old behavior.
-                line = line.decode('utf-8').rstrip('\r\n')
+                line = line.rstrip(b'\r\n')
                 log.info('Data: %r', line)
                 if not line:
                     yield from self.push('500 Error: bad syntax')
                     continue
-                i = line.find(' ')
+                i = line.find(b' ')
+                # Decode to string only the command name part, which must be
+                # ASCII as per RFC.  If there is an argument, it is decoded to
+                # UTF-8/surrogateescape so that non-UTF-8 data can be
+                # re-encoded back to the original bytes when the SMTP command
+                # is handled.
                 if i < 0:
-                    command = line.upper()
+                    command = str(line.upper(), encoding='ascii')
                     arg = None
                 else:
-                    command = line[:i].upper()
+                    command = str(line[:i].upper(), encoding='ascii')
```

Comment:
I don't think it matters in practice but I don't like that we're inconsistent about the use of `str(s, encoding=c)` and `.decode()`/`.encode()`.  I'd like to at least be consistent and I'm +0 about using `.decode()`/`.encode()` since I think it's a bit easier to search/grep for.